psydac/linalg/block.py

```diff
@@ -264,6 +270,8 @@ def dot( self, v, out=None ):
         else:
             out = BlockVector( self._codomain )
 
+        v.update_ghost_regions()
```

Comment:
Don't the linear operators `Lij` below call this method on each component `v[j]` already?

Reply:
yes for some raison it locks if we do it in each component and this solves the problem, we also want to separate the communications from the calculations  to be able to optimize them in the future

Reply:
I am worried about this MPI deadlock: it should not occur unless we have a serious bug, which was maybe introduced by this PR, or had been here before and remained undetected for a very long time. We must investigate this.

Reply:
After some discussion we solved the mystery: only some processes in the MPI communicator have an `InterfaceLinearOperator` as an off-diagonal block in the block matrix. When the `dot` method of this operator is called on a vector `v` with `v.ghost_regions_in_sync == False` the method `v.update_ghost_regions()` is called, but not all processes contribute to this collective call. This results in a deadlock.

Reply:
Since `InterfaceLinearOperator` cannot safely call the `update_ghost_regions` method of a `StencilVector` object `v`, I recommend that it raises an error (MPI abort) if it finds that `v.ghost_regions_in_sync == False`.